src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -960,26 +959,26 @@ void TaskSketcherElements::slotElementsChanged(void)
 
     ui->listWidgetElements->clear();
 
+    using GeometryState = ElementItem::GeometryState;
+
     int i=1;
     for(std::vector< Part::Geometry * >::const_iterator it= vals.begin();it!=vals.end();++it,++i){
         Base::Type type = (*it)->getTypeId();
+        GeometryState state = GeometryState::Normal;
+
         bool construction = Sketcher::GeometryFacade::getConstruction(*it);
         bool internalAligned = Sketcher::GeometryFacade::isInternalAligned(*it);
-        if (internalAligned) 
-            construction = false;
 
-        ElementWidgetIcons::Type icontype = construction ? ElementWidgetIcons::Type::Construction : (internalAligned ? ElementWidgetIcons::Type::Internal : ElementWidgetIcons::Type::Normal);
+        if (construction && !internalAligned)
+            state = GeometryState::Construction;
+        else if (internalAligned)
+            state = GeometryState::InternalAlignment;

```

Comment:
```
if (internalAligned)
    state = GeometryState::InternalAlignment;
else if (construction)
    state = GeometryState::Construction
```
:)

Reply:
This code looks better but it hides the fact that order matter. If someone comes after maybe it's not obvious that internalAligned are also construction so he may make msitake. No?

Reply:
The original code does not make it clear either. It just look like suboptimal coding, hence my remark. ;) 
If you think it is an important or helpful info, comments are your friends. :)

Reply:
I am not following this one.

The GeometryState of the widget item is a mutually exclusive state (because one cannot classify for icons, strings, a geometry in two different categories at the same time). Then a single enum is a better option which, on top of avoiding combinations, prevents boolean blindness (those three in a row booleans that were before in the item constructor, which could easily be swapped by mistake). As it is a enum class, there is a strong typing check by the compiler preventing implicit conversions.

The State of the sketcher geometry can be construction and internal alignment at the same time, as internal alignment is broadly treated as construction (e.g. does not define edges out of edit mode), with of course certain specific behaviour where applicable.

For building the Item we need to extract information from the sketch and handle the non-mutual exclusivity of sketcher booleans into the enum.

Perhaps there is something obvious I do not see?

Reply:
@0penBrain yes you are correct just a small comment is better
@abdullahtahiriyo I'm not sure what you mean. 
Should I commit the change offered by OpenBrain or are you active on the branch already Abdullah and will handle it on your side?